include/swift/SILOptimizer/Utils/InstOptUtils.h

```diff
@@ -571,6 +572,17 @@ bool optimizeMemoryAccesses(SILFunction &fn);
 /// See the PredictableDeadAllocationElimination pass.
 bool eliminateDeadAllocations(SILFunction &fn);
 
+/// Involves a synchronization point like a memory barrier, lock or syscall.
```

Comment:
We don't need this in C++, because all clients of this API in C++ should use the inter-procedural-aware API `Instruction.isDeinitBarrier(_ analysis: CalleeAnalysis)` anyway (which is bridged in CalleeAnalysis).